Jellyfin.Server.Implementations/Item/BaseItemRepository.cs

```diff
@@ -1040,41 +1078,71 @@ public static BaseItemDto DeserialiseBaseItem(BaseItemEntity baseItemEntity, ILo
+            var musicArtistTypeName = _itemTypeLookup.BaseItemKindNames[BaseItemKind.MusicArtist];
+            var audioTypeName = _itemTypeLookup.BaseItemKindNames[BaseItemKind.Audio];
+            var trailerTypeName = _itemTypeLookup.BaseItemKindNames[BaseItemKind.Trailer];
+
+            var resultQuery = query.Select(e => new
             {
-                SeriesCount = e.ItemValues!.Count(f => f.Item.Type == seriesTypeName),
-                EpisodeCount = e.ItemValues!.Count(f => f.Item.Type == episodeTypeName),
-                MovieCount = e.ItemValues!.Count(f => f.Item.Type == movieTypeName),
-                AlbumCount = e.ItemValues!.Count(f => f.Item.Type == musicAlbumTypeName),
-                ArtistCount = e.ItemValues!.Count(f => f.Item.Type == musicArtistTypeName),
-                SongCount = e.ItemValues!.Count(f => f.Item.Type == audioTypeName),
-                TrailerCount = e.ItemValues!.Count(f => f.Item.Type == trailerTypeName),
-            }
-        });
+                item = e.AsQueryable()
+                        .Include(e => e.TrailerTypes)
+                        .Include(e => e.Provider)
+                        .Include(e => e.LockedFields)
+                        .Include(e => e.Images)
+                        .AsSingleQuery().First(),
+                // TODO: This is bad refactor!
+                itemCount = new ItemCounts()
+                {
+                    SeriesCount = itemCountQuery!.Count(f => f.Type == seriesTypeName),
+                    EpisodeCount = itemCountQuery!.Count(f => f.Type == episodeTypeName),
+                    MovieCount = itemCountQuery!.Count(f => f.Type == movieTypeName),
+                    AlbumCount = itemCountQuery!.Count(f => f.Type == musicAlbumTypeName),
+                    ArtistCount = itemCountQuery!.Count(f => f.Type == musicArtistTypeName),
+                    SongCount = itemCountQuery!.Count(f => f.Type == audioTypeName),
+                    TrailerCount = itemCountQuery!.Count(f => f.Type == trailerTypeName),
+                }
+            });
 
-        result.StartIndex = filter.StartIndex ?? 0;
-        result.Items = resultQuery.ToArray().Where(e => e is not null).Select(e =>
+            result.StartIndex = filter.StartIndex ?? 0;
+            result.Items = [.. resultQuery.ToArray().Where(e => e is not null).Select(e =>
+            {
+                return (DeserialiseBaseItem(e.item, filter.SkipDeserialization), e.itemCount);
+            })];
+        }
+        else
         {
-            return (DeserialiseBaseItem(e.item, filter.SkipDeserialization), e.itemCount);
-        }).ToArray();
+            result.StartIndex = filter.StartIndex ?? 0;
+            result.Items = [.. query.Select(e => e.First()).ToArray().Where(e => e is not null).Select<BaseItemEntity, (BaseItemDto, ItemCounts?)>(e =>
+            {
+                return (DeserialiseBaseItem(e, filter.SkipDeserialization), null);
+            })];
```

Comment:
```suggestion
            result.Items =
            [
                .. query.Select(e => e.First())
                    .AsEnumerable()
                    .Where(e => e is not null)
                    .Select<BaseItemEntity, (BaseItemDto, ItemCounts?)>(e => (DeserialiseBaseItem(e, filter.SkipDeserialization), null))
            ];
```

No reason to to `.ToArray()` here, right?
Rider seems to think the formatting will be a future warning

Reply:
you _could_ do AsEnumerable here instead i guess